test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -5711,5 +5711,220 @@ public void Dotnet_New_Template_Restore_Pack_Success(string template)
+    <PackageReference Include=""{stableDependencyName}"" Version=""{stableDependencyVersion}""/>
+    <PackageReference Include=""{prereleaseDependencyName}"" Version=""{prereleaseDependencyVersion}"" NoWarn = ""NU5104""/>
+  </ItemGroup>
+</Project>";
+                File.WriteAllText(projectFile, projectXml);
+
+                // Act
+                msbuildFixture.RestoreProject(workingDirectory, projectName, string.Empty);
+                CommandRunnerResult result = msbuildFixture.PackProject(workingDirectory, projectName, $"/p:PackageOutputPath={workingDirectory}", validateSuccess: false);
+
+                // Assert
+                result.Success.Should().BeTrue();
+                var nupkgPath = Path.Combine(workingDirectory, $"{projectName}.1.2.3.nupkg");
+                var nuspecPath = Path.Combine(workingDirectory, "obj", $"{projectName}.1.2.3.nuspec");
+                Assert.True(File.Exists(nupkgPath), "The output .nupkg is not in the expected place");
+                result.AllOutput.Should().NotContain(stableDependencyName);
+                result.AllOutput.Should().NotContain(prereleaseDependencyName);
+                result.AllOutput.Should().NotContain(NuGetLogCode.NU5104.ToString());
+
+                using (var nupkgReader = new PackageArchiveReader(nupkgPath))
+                {
+                    var nuspecReader = nupkgReader.NuspecReader;
+
+                    var dependencyGroups = nuspecReader
+                        .GetDependencyGroups()
+                        .OrderBy(x => x.TargetFramework,
+                            new NuGetFrameworkSorter())
+                        .ToList();
+
+                    Assert.Equal(1,
+                        dependencyGroups.Count);
+
+                    var dependencyPackage = dependencyGroups[0].Packages.ToList();
+                    Assert.Equal(2, dependencyPackage.Count);
+                    Assert.Equal(prereleaseDependencyName, dependencyPackage[0].Id);
+                    Assert.Equal(new VersionRange(new NuGetVersion(prereleaseDependencyVersion), true, null, true), dependencyPackage[0].VersionRange);
+                    Assert.Equal(stableDependencyName, dependencyPackage[1].Id);
+                    Assert.Equal(new VersionRange(new NuGetVersion(stableDependencyVersion), true, null, true), dependencyPackage[1].VersionRange);
+                    Assert.Equal(new List<string> { "Analyzers", "Build" }, dependencyPackage[0].Exclude);
+                    Assert.Equal(new List<string> { "Analyzers", "Build" }, dependencyPackage[1].Exclude);
+                    Assert.Empty(dependencyPackage[0].Include);
+                    Assert.Empty(dependencyPackage[1].Include);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task PackCommand_PackProject_PackageReference_PreReleaseDependency_ProjectLevelNoWarn_Succeed()
+        {
+            using (var pathContext = msbuildFixture.CreateSimpleTestPathContext())
```

Comment:
I didn't add mutlitarget frameworks test due to limitations imposed by [msbuildFixture.CreateSimpleTestPathContext())](https://github.com/nuget/client.engineering/issues/1347). For multitarget framework I manually tests on my local several times and it works.

Reply:
There are other tests in this exact same class that multi target.

Reply:
They use `TestDirectory.Create()`, not msbuildFixture.CreateSimpleTestPathContext(). If I use `TestDirectory.Create()` then unit test makes call to nuget.org which makes test more flaky.

Reply:
> They use `TestDirectory.Create()`, not msbuildFixture.CreateSimpleTestPathContext(). If I use `TestDirectory.Create()` then unit test makes call to nuget.org which makes test more flaky.

I'm not sure how that's relevant. 
To create a multi targetting project you need to edit the project file, which is not affected by whether you used CreateSimpleTestPathContext vs TestDirectory.Create(). 

There's no generic problem with TestDirectory vs SimpleTestPathContext. If it ends up calling to nuget.org, it's because the test is setup that way, nothing to do with the multi targetting.

Reply:
I [already tried it](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5562335&view=ms.vss-test-web.build-test-results-tab&runId=30423028&resultId=110124&paneView=debug), and I got following error, it looks it's trying to restore packages from nuget.org for frameworks other than net5.0

`error NU1101: Unable to find package Microsoft.NETFramework.ReferenceAssemblies. No packages exist with this id in source(s): source, template`

[Here is setup for project I used for above](https://github.com/NuGet/NuGet.Client/pull/4369/commits/06e479915439200226942441f720d91721d958d9).

```
 tring projectXml = $@"<Project Sdk=""Microsoft.NET.Sdk"">
  <PropertyGroup>
    <TargetFrameworks>net5.0;net451</TargetFrameworks>
    <Version>1.2.3</Version>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include=""{stableDependencyName}"" Version=""{stableDependencyVersion}""/>
    <PackageReference Include=""{prereleaseDependencyAName}"" Version=""{prereleaseDependencyAVersion}"" NoWarn = ""NU5104""/>
  </ItemGroup>
```

Reply:
Can you disable the reference assembly package download?

Reply:
Do we have example? I don't know how to stop download package.

Reply:
I generated the preprocessed msbuild file via `/pp:pp.txt` and searching for the package id I found: 

```xml
  <Target Name="IncludeTargetingPackReference" BeforeTargets="_CheckForInvalidConfigurationAndPlatform;CollectPackageReferences;CheckForImplicitPackageReferenceOverrides" Condition="'$(TargetFrameworkMoniker)' != '' and '$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(AutomaticallyUseReferenceAssemblyPackages)' == 'true'">
    <GetReferenceAssemblyPaths TargetFrameworkMoniker="$(TargetFrameworkMoniker)" RootPath="$(TargetFrameworkRootPath)" TargetFrameworkFallbackSearchPaths="$(TargetFrameworkFallbackSearchPaths)" BypassFrameworkInstallChecks="true" SuppressNotFoundError="true">
      <Output TaskParameter="FullFrameworkReferenceAssemblyPaths" PropertyName="_FullFrameworkReferenceAssemblyPaths" />
    </GetReferenceAssemblyPaths>
    <ItemGroup>
      <_ExistingReferenceAssembliesPackageReference Include="@(PackageReference)" Condition="'%(PackageReference.Identity)' == 'Microsoft.NETFramework.ReferenceAssemblies'" />
      <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="$(MicrosoftNETFrameworkReferenceAssembliesLatestPackageVersion)" IsImplicitlyDefined="true" PrivateAssets="All" Condition="'$(_FullFrameworkReferenceAssemblyPaths)' == '' and '@(_ExistingReferenceAssembliesPackageReference)' == ''" />
    </ItemGroup>
  </Target>
```

Try `AutomaticallyUseReferenceAssemblyPackages`

I wonder if the build fails.

Reply:
I changed my test as below, but got different kind of error on CI.
```
  <PropertyGroup>
    <TargetFrameworks>net5.0;net451</TargetFrameworks>
    <Version>1.2.3</Version>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <AutomaticallyUseReferenceAssemblyPackages>false</AutomaticallyUseReferenceAssemblyPackages>
  </PropertyGroup>

```

`The reference assemblies for .NETFramework,Version=v4.5.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks `

Reply:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5564757&view=ms.vss-test-web.build-test-results-tab&runId=30452534&resultId=110112&paneView=debug

Reply:
VS 2022 no longer provides .NET Framework installers for old, unsupported, versions of the .NET Framework, like 4.5.1.  Instead, use a version of the .NET Framework that is supported/available via the VS 2022 installer, such as `net48`.

Reply:
Thank you for your suggestion.  But [problem still persists](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5565494&view=ms.vss-test-web.build-test-results-tab&runId=30464206&resultId=110125&paneView=debug).

`error MSB3644: The reference assemblies for .NETFramework,Version=v4.8 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks `

Reply:
@nkolev92 @zivkan 
It looks Karteek added escape hatch for similar issue. May I use it?

https://github.com/NuGet/NuGet.Client/blob/835733dec384619472a23a47057363ddbc705b1d/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestSettingsContext.cs#L182-L183